### PR TITLE
Document the usage of transaction for the stream() methods

### PR DESCRIPTION
**Pull request by loicmathieu**
https://github.com/quarkusio/quarkus/issues/3256

### CODE REVIEW

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc [hunk 1]
```diff
@@ -188,6 +188,9 @@ List<String> namesButEmmanuels = persons
     .collect(Collectors.toList());
 --
 
+NOTE: The `stream` methods requires a transaction to works properly.
```
Comment (machi1990) [inline docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]:
```
NOTE: The `stream` method requires a transaction to work properly.
```

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java [hunk 2]
```diff
@@ -373,6 +373,8 @@ public void flush() {
     /**
      * Find entities matching a query, with optional indexed parameters.
      * This method is a shortcut for <code>find(query, params).stream()</code>.
+     * It requires a transaction to works properly.
```
Comment (machi1990) [inline extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java]:
```
It requires a transaction to work properly.
```

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc (hunk 1 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: There is multiple `stream` methods

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java (hunk 2 above)
Comment (loicmathieu) [reply extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java]: done

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc [hunk 3]
```diff
@@ -188,6 +188,9 @@ List<String> namesButEmmanuels = persons
     .collect(Collectors.toList());
 --
 
+NOTE: The `stream` methods requires a transaction to work properly.
```
Comment (kenfinnigan) [inline docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: Change "requires" to "require"

Comment (kenfinnigan) [reply docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: Maybe also drop "properly"? Either it works or it doesn't

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java [hunk 4]
```diff
@@ -373,6 +373,8 @@ public void flush() {
     /**
      * Find entities matching a query, with optional indexed parameters.
      * This method is a shortcut for <code>find(query, params).stream()</code>.
+     * It requires a transaction to work properly.
```
Comment (kenfinnigan) [inline extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java]: "Requires a transaction to work", and apply to all the same comments in the classes

**Review by kenfinnigan — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc (hunk 3 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: done

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java (hunk 4 above)
Comment (loicmathieu) [reply extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntityBase.java]: done

**Comment by loicmathieu** [comment]
@kenfinnigan I updated the documentation as you requested.

**Review by kenfinnigan — APPROVED** [review]

**Comment by kenfinnigan** [comment]
Thanks @loicmathieu

**Pull request merged**